types/document.d.ts

```diff
@@ -287,6 +287,10 @@ declare module 'mongoose' {
     toJSON(options: ToObjectOptions & { flattenMaps: true, virtuals: true }): FlattenMaps<Require_id<DocType & TVirtuals>>;
     toJSON(options: ToObjectOptions & { flattenMaps: true }): FlattenMaps<Require_id<DocType>>;
 
+    // flattenUUIDs overloads
+    toJSON(options: ToObjectOptions & { flattenUUIDs: true, virtuals: true }): UUIDToString<Require_id<DocType & TVirtuals>>;
```

Comment:
Unfortunately we would also need to handle all other combinations:

- `flattenUUIDs: true, flattenMaps: true`
- `flattenUUIDs: true, flattenObjectIds: true`
- `flattenUUIDs: true, flattenObjectIds: true, flattenMaps: true`
- `flattenUUIDs: true, flattenObjectIds: true, flattenMaps: true, virtuals: true`
- `flattenUUIDs: true, flattenObjectIds: true, virtuals: true`
- `flattenUUIDs: true, flattenMaps: true, virtuals: true`

Plus versionKey variants for completeness' sake.

Honestly this is why I don't like adding new options to toObject() and toJSON() unless absolutely necessary: they cause combinatorial explosion in TypeScript. There isn't a good way to handle all the different cases without explicitly listing out every single override.